File: app/src/main/java/com/neema/shifumi/MainActivity.java
package com.neema.shifumi;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.neema.shifumi.no_android.Player;
import com.neema.shifumi.no_android.PlayerType;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = MainActivity.class.getName();
    private boolean exitApp;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        exitApp = false;

        Button vsHuman = findViewById(R.id.btn_vs_human);
        Button vsComputer = findViewById(R.id.btn_vs_computer);

        final Intent intent = new Intent(MainActivity.this, PlayActivity.class);

        vsHuman.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                intent.putExtra("VS_COMPUTER", false);
                intent.putExtra(Player.class.getName(), new Player(PlayerType.HUMAN));
                startActivity(intent);
                finish();
            }
        });

        vsComputer.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                intent.putExtra(Player.class.getName(), new Player(PlayerType.COMPUTER));
                intent.putExtra("VS_COMPUTER", true);
                startActivity(intent);
                finish();
            }
        });
    }

    /*@Override
    public void onBackPressed() {
        if (exitApp) {
            super.onBackPressed();
            return;
        }
        this.exitApp = true;
        Toast.makeText(this, "Veuillez cliquer de nouveau sur BACK pour quitter", Toast.LENGTH_SHORT).show();
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                exitApp = false;
            }
        }, 1000);
    }*/
}

File: app/src/main/java/com/neema/shifumi/MainActivity.kt
package com.neema.shifumi

import android.content.Intent
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.neema.shifumi.gameclass.Computer
import com.neema.shifumi.gameclass.Human
import kotlinx.android.synthetic.main.activity_main.*
import kotlin.system.exitProcess

class MainActivity : AppCompatActivity() {

    companion object {
        private val TAG = MainActivity::class.java.simpleName
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d(TAG, "------------------ $TAG OnCreate ------------------ ")
        setContentView(R.layout.activity_main)

        val intent = Intent(this@MainActivity, GameActivity::class.java)

        btn_vs_computer.setOnClickListener {
            Log.d(TAG, "button ${Human::class.java.simpleName} vs ${Computer::class.java.simpleName} ...")
            intent.putExtra("VS_COMPUTER", true)
            startActivity(intent)
        }

        btn_only_computer.setOnClickListener {
            Log.d(TAG, "button ${Computer::class.java.simpleName} vs ${Computer::class.java.simpleName} ...")
            intent.putExtra("VS_COMPUTER", false)
            startActivity(intent)
        }
    }

    private fun showExitDialog() {

        // before inflating the custom alert dialog layout, we will get the current activity viewGroup
        val viewGroup = findViewById<ViewGroup>(android.R.id.content)
        val dialogView = LayoutInflater.from(this).inflate(R.layout.alert_dialog_ask, viewGroup, false)

        val builder = AlertDialog.Builder(this)
        builder.setCancelable(false)
        builder.setView(dialogView)

        val alertDialog = builder.create()
        alertDialog.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT)) // Supprime le background
        val title = dialogView.findViewById<TextView>(R.id.tv_title_dialog)
        val tvVS = dialogView.findViewById<TextView>(R.id.tv_type_party)
        title.text = getString(R.string.exit_game)
        tvVS.text = getString(R.string.exit_game_ask)
        val yep = dialogView.findViewById<Button>(R.id.btn_yep)
        val nope = dialogView.findViewById<Button>(R.id.btn_nope)
        yep.setOnClickListener {
            alertDialog.dismiss()
            finish()
            exitProcess(0)//android.os.Process.killProcess(android.os.Process.myPid())
        }
        nope.setOnClickListener {
            alertDialog.dismiss()
        }
        alertDialog.show()

    }

    override fun onBackPressed() {
        showExitDialog()
    }
}
